### DISCUSSION TITLE
How to verify that a specific plugin is installed and/or activated?

### DISCUSSION THREAD

**Discussion by jeffshead** [YOURLS/YOURLS#3476] [Q&A] [answered]
I searched but did not find an answer.

In a separate plugin, I would like to display a link but only if the plugin named Ip-Viewer (https://github.com/faab007nl/Yourls-Ip-Viewer) is installed and activated.

For example:

    if ( the Ip-Viewer plugin is installed && activated ) {
        echo "<a href='" . $yourls_site . "/admin/plugins.php?page=ip-viewer&shortcode='" . $keyword . "'>View IP log</a>";
    } else {
        echo 'You need to install and activate the Ip-Viewer plugin!';
    }

What is the correct, actual code to use for the **if** statement?

**Comment by dgw** [comment, marked as answer]
You will need to know the full path to the plugin file. These functions can help get the available plugins (with info) and check if a specific plugin is active:

https://github.com/YOURLS/YOURLS/blob/6da30dfab08700888d4403a8ce2abaaf47ec11c7/includes/functions-plugins.php#L483-L511

**Comment by jeffshead** [comment]
Unbelievable!!!

I searched **functions-plugins.php** a couple three times and overlooked `yourls_is_active_plugin()` every time.

Thank you, DGW.
